lib/ExpensiMark.js

```diff
@@ -49,6 +49,38 @@ export default class ExpensiMark {
                 replacement: '$1<code>$2</code>$3',
             },
 
+            /**
+             * Apply the hereMention first because the string @here is still a valid mention for the userMention regex.
+             * This ensures that the hereMention is always considered first, even if it is followed by a valid userMention.
+            */
+            {
+                name: 'hereMentions',
+                regex: /(?<![^\s]+)@here\b/g,
+                replacement: match => `<mention-here>${match}</mention-here>`,
+            },
+
+            /**
+             * This regex matches a valid user mention in a string.
+             * A user mention is a string that starts with the '@' symbol and is followed by a valid user's primary login
+             * The username can contain any combination of letters, numbers, periods, underscores, and hyphens
+             *
+             * Note: currently we are only allowing mentions in a format of @+19728974297@expensify.sms and @username@expensify.com
+             * inorder to match mentions in a format of @username we should update the regex to: /(?!@here)(?<![^_~#*\s]+)@([\w.@+-]+(?:\.[\w@+-]+)*\b)/g
+             */
+            {
+                // (?<![^~#*-]+) excludes every mention with a character infront except ~, #, * and +
+                // this is to allow bold, italic, strikethrough, quote, heading1 on a mention
+                name: 'userMentions',
+                regex: /(?!@here)(?<![^_~#*\s]+)@([\w.@+-]+@+\w+[.]+\w+(?:\.[\w@+-]+)*\b)/g,
+                replacement: (match) => {
+                    if (match[match.length - 1] === '_') {
```

Comment:
could you kindly explain what does this doing? why do we need to account for situations in which the email address ends with `_`? Is this for italic?